pkg/metrics/client_go_adapter.go

```diff
@@ -37,20 +39,30 @@ var (
 		},
 		[]string{"code", "method", "host"},
 	)
+
+	clientMetricsRegisterOnce = sync.Once{}
 )
 
 func init() {
-	registerClientMetrics()
-}
-
-// registerClientMetrics sets up the client latency metrics from client-go.
-func registerClientMetrics() {
 	// register the metrics with our registry
 	Registry.MustRegister(requestResult)
 
-	// register the metrics with client-go
-	clientmetrics.Register(clientmetrics.RegisterOpts{
-		RequestResult: &resultAdapter{metric: requestResult},
+	time.AfterFunc(30*time.Second, func() {
```

Comment:
I am not a fan of this because it means you might not get metrics for the first thirty seconds of the binary. Another way I can think of is to maybe call this before first use in the controller or so, that would still allow to register them yourselves by doing it before you build your controllers and avoid this issue?

Reply:
Agreed, I moved this call into the NewServer function of metric server in 639312c0e24c606365d41c98d95d163913a209fc